modules/ding_webtrekk/ding_webtrekk.module

```diff
@@ -51,24 +46,92 @@ function ding_webtrekk_page_alter(&$page) {
+
+  // Tracks whether the user is logged in and, if logged in, what login provider
+  // was used.
+  if (user_is_logged_in()) {
+    // Note that this separation of logged in users might be problematic if some
+    // pages are also cached for authenticated users.
+    if (module_exists('ding_gatewayf') && _ding_gatewayf_is_logged_in_with_gatewayf()) {
+      $params['p_isloggedin'] = 'NemId';
+
+      // This is the landing page for successful user registration with gatewayf.
+      // If the registration was successful, the users will be automatically
+      // logged in. So by also requiring a logged in user before setting parameter
+      // we reduce the chances of false positives. For example, it will prevent
+      // random access to the URL from anonymous users, from being tracked.
+      if (strpos(current_path(), DING_GATEWAYF_REGISTRATION_SUCCESS_URL) === 0) {
+        $params['p_usercreate'] = 'NemId';
+      }
+    }
+    // Fallback to cpr+pin if none the modules we know about was used to log in.
+    else {
+      $params['p_isloggedin'] = 'cpr+pinkode';
+    }
+
+  }
+
   // Track ting search results. We take advantage of the fact that ting_search
   // module stores the result in a static variable when search is triggered via
   // search_data() function from core search module. Since this is not tied to
   // any path, we also check current_path().
   if (strpos(current_path(), 'search/ting') === 0) {
     if ($search_result = ting_search_current_results()) {
-      $params = [
-        'pageTitle' => 'Search ting',
+      $params += [
         'OSS' => $search_result->getSearchRequest()->getFullTextQuery(),
         'OSSr' => $search_result->getNumTotalObjects(),
       ];
-      _ding_webtrekk_set_data_layer_values($params);
+
+      // We want to track facets that users have added themselves from the UI.
+      // Looking for the special URL parameter that is added in these cases
+      // seems to be more reliable than looking on the filters added to the
+      // search request object.
+      if (!empty($_GET['facets']) && is_array($_GET['facets'])) {
+        $params['p_s_Facets'] = implode(';', $_GET['facets']);
+      }
+
+      // Track the page of the search result.
+      $params['p_s_Page'] = $search_result->getSearchRequest()->getPage();
+
+      // Track the sort used in the search request.
+      if (!empty(ting_search_current_results()->getSearchRequest()->getSorts())) {
+        $current_sort = $search_result->getSearchRequest()->getSorts()[0];
+        $params['p_s_Sort'] = ting_search_sort_key_from_sort($current_sort);
+      }
+      else {
+        $default_sort = variable_get('ting_search_default_sort', '');
+        // If empty string the default "sort" is ranking. If search provider is
+        // opensearch we know how to get the actual ranking used. If not we will
+        // use a generic 'rank' value for the parameter.
+        if ($default_sort === '') {
+          $params['p_s_Sort'] = 'rank';
```

Comment:
Why not just set the default value for `ting_search_default_sort` to `rank`:

```php
$default_sort = variable_get('ting_search_default_sort', 'rank');
```

Reply:
Good point. Will fix this.